src/plugins/console/public/application/containers/editor/monaco/hooks/use_register_keyboard_commands.ts

```diff
@@ -0,0 +1,160 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useRef } from 'react';
+import { i18n } from '@kbn/i18n';
+import { monaco } from '@kbn/monaco';
+// eslint-disable-next-line @kbn/eslint/module_migration,@kbn/imports/no_unresolvable_imports
+import { IDisposable } from 'monaco-editor';
+
+interface RegisterKeyboardCommandsParams {
+  /** The current Monaco editor instance. */
+  editor: monaco.editor.IStandaloneCodeEditor;
+  /** Function for sending the selected request(s). */
+  sendRequest: () => void;
+  /** Function for indenting the selected request(s). */
+  autoIndent: () => void;
+  /** Function that returns the documentation link for the selected request. */
+  getDocumentationLink: () => Promise<string | null> | undefined;
+  /** Function for moving the cursor to the previous request edge. */
+  moveToPreviousRequestEdge: () => void;
+  /** Function for moving the cursor to the next request edge. */
+  moveToNextRequestEdge: () => void;
+}
+
+const SEND_REQUEST_ACTION_ID = 'sendRequest';
+const AUTO_INDENT_ACTION_ID = 'autoIndent';
+const OPEN_DOCS_ACTION_ID = 'openDocs';
+const MOVE_UP_ACTION_ID = 'moveUp';
+const MOVE_DOWN_ACTION_ID = 'moveDown';
+const MOVE_TO_LINE_ACTION_ID = 'moveToLine';
+
+/**
+ * Hook that returns a function for registering keyboard commands in the editor.
+ *
+ * @param params The {@link RegisterKeyboardCommandsParams} to use.
+ */
+export const useKeyboardCommandsUtils = () => {
+  const sendRequestAction = useRef<IDisposable | null>(null);
+  const autoIndentAction = useRef<IDisposable | null>(null);
+  const openDocsAction = useRef<IDisposable | null>(null);
+  const moveToPreviousAction = useRef<IDisposable | null>(null);
+  const moveToNextAction = useRef<IDisposable | null>(null);
+  const moveToLineAction = useRef<IDisposable | null>(null);
+
+  const disposeAllActions = () => {
+    if (sendRequestAction.current) {
+      sendRequestAction.current.dispose();
+    }
+    if (autoIndentAction.current) {
+      autoIndentAction.current.dispose();
+    }
+    if (openDocsAction.current) {
+      openDocsAction.current.dispose();
+    }
+    if (moveToPreviousAction.current) {
+      moveToPreviousAction.current.dispose();
+    }
+    if (moveToNextAction.current) {
+      moveToNextAction.current.dispose();
+    }
+    if (moveToLineAction.current) {
+      moveToLineAction.current.dispose();
+    }
+  };
+
+  const registerKeyboardCommands = (params: RegisterKeyboardCommandsParams) => {
+    const {
+      editor,
+      sendRequest,
+      autoIndent,
+      getDocumentationLink,
+      moveToPreviousRequestEdge,
+      moveToNextRequestEdge,
+    } = params;
+
+    const openDocs = async () => {
+      const documentation = await getDocumentationLink();
+      if (!documentation) {
+        return;
+      }
+      window.open(documentation, '_blank');
+    };
+
+    disposeAllActions();
```

Comment:
why do we need to call this function here?

Reply:
I called it here just in case there are already any actions, so we can dispose them before we create new ones. Similarly to what we did for the resize checker (https://github.com/elastic/kibana/pull/180494#discussion_r1579180997).